### ISSUE TITLE
[Feature request] Add support to creat/edit/delete/"delete all"/drage & drope outlines of PDF

### ISSUE THREAD

**Issue by Nokia808** [jeromerobert/pdfarranger#235]
Hi.

pikepdf already added this feature, though may be not so completed as we wish, so it will be very useful if you add these functionalities to pdfarranger as currently no open source GUI application available to perform them on Linux.

If you add this feature, then it will be a great jump forward for your application.

**Comment by jeromerobert** [comment]
> pdfpike

pikepdf I guess

> already added this feature

Where ? Do you have a link which explain that ?

**Comment by Nokia808** [comment]
@jeromerobert 
Yes I meant pikepdf. I will correct my 1st post in this issue.

Kindly, see this:
https://pikepdf.readthedocs.io/en/latest/
Specifically for outlines - see:
https://pikepdf.readthedocs.io/en/latest/topics/outlines.html

But please if you implement this very wanted feature, kindly, not only an outline to page, but it should be WITHIN the page also & it should support tree arrangement & drag & drop & "select all for deletion purpose"  ..... 

See the PR for this feature that already is merged:
https://github.com/pikepdf/pikepdf/pull/82
As you see, there is certain limitations, like inability to change "font" & "colour" of outline ... However, these limitations are not basic & can be addressed in feature by pikepdf developer ...

**Comment by Nokia808** [comment]
@jeromerobert 
I like to add the following note: to implement this feature you need special design, so that when new option of "Manage Outlines" clicked by user then UI of pdfarranger should include side bar [see red arrow in screenshot] to show & manage outlines, & main area [ see blue circle in screenshot] where content of a page will appear, & user can scroll pages up & down in "continuous view mode", like that in Okular. I mean by this: when user click "Manage Outlines" option, then current "thumbnails" view should be changed into "continuous page view with vertical scroll ability".

Also, user should have option to "Close Outlines Management", so that when use this option "Close Outlines Management" the "continuous page view" will closed & return back to "thumbnails view". Off-course this should asked the user to save or not the changes if she/he not already saved them.

![example](https://user-images.githubusercontent.com/25375330/88463084-af09d300-ce9f-11ea-82d9-e14b596747ee.png)

**Comment by jeromerobert** [comment]
Now that i understand what you want I have to say that it is far beyond pdfarranger goal (see pdfarranger Github About). That would require a huge amont of work, probably a lot of rewrite. I won't do it, won't maintain it and as this would be highly intrusive in existing code won't even accept it as a contribution (which I doubt it would ever get there).

If a pdfarranger developer thinks I missed something and that, it can be implemented and maintained with a reasonable amount of work, I'll gladly reopen that bug.

**Comment by Nokia808** [comment]
@jeromerobert 
I was expected this, because - opposite to "PDF Mix Tool" which is at earlier step in it's development - your program in more advanced stage of it's development (has already more features), so it will be more difficult to you to change every thing ....

To me, I see Okular is the best place for this feature, but they not react with my request, may be due to technical causes - see the following:
https://bugs.kde.org/show_bug.cgi?id=377454
Okular is already has excellent & nearly complete annotation tools & it will full a big gap if it include this feature, but .......

**Comment by Nokia808** [comment]
@jeromerobert 
@dreua 
Dear, before dictating this comment, kindly appreciate the fact that I'm not a programmer. So, I see the picture from outside not from inside. For that my idea & comment some time correct & other time incorrect, incorrect due to lack of technical knowledge ....

I read the following issue https://github.com/jeromerobert/pdfarranger/issues/51
From it I discovered that ZOOM level is already saved !! I examined it & it working very well ! For example see the following 3 screenshots for same PDF:

1st screenshot is with thumbnail zoom level:

![1](https://user-images.githubusercontent.com/25375330/89637033-1c066b00-d899-11ea-8471-582e0505888d.png)

2nd screenshot after "zoom in" for further 5 level bigger, at the zoom level when only SINGLE page appear within screen - kindly see the side EMPTY areas where I put a red elliptical shape to mark :

![2](https://user-images.githubusercontent.com/25375330/89637163-4a844600-d899-11ea-8ba4-b653e2cfd9ee.png)
 
3rd screenshot is same of 2nd screenshot, but it show that user CAN scroll it (I scrolled it down):

![3](https://user-images.githubusercontent.com/25375330/89637499-ca121500-d899-11ea-8759-9f95878fad80.png)

Now dear developers, are these what we needed as UI to implement this closed issue ? This what seem to me - correct to me if wrong:
- zoom level of page to 1ST zoom which make single page appear within UI,
- user can scroll pages within UI
- empty area(s) to add within it(them) UI for create/delete/edit outlines.

It seem to me that all what needed is co-ordinate new option "Outlines management" with zoom level, so that when user click on proposed "Outlines management" the program will automatically zoom EACH pages to 1st zoom level which make it the only one appear within UI, so that empty area(s) will be existing for UI of outlines management option & user can scroll the page & work on outlines .....

The above idea may be not what needed exactly but a modification from it is the exactly needed. Kindly, see it, probably it is of use .......
